test/test_eager_transforms.py

```diff
@@ -1198,20 +1247,58 @@ def f(x):
         self.assertEqual(result, torch.eye(3, 3, device=device))
         self.assertEqual(aux, x.cos())
 
-    @FIXME_jacrev_only
+    @jacrev_and_jacfwd
     def test_aux_pytree(self, device, jacapi):
         def f(x):
             y = x.clone()
             return y, {'a': y.cos(), 'b': [y.tan()]}
 
         x = torch.randn(3, device=device)
-        result, aux = jacapi(f, has_aux=True)(x)
 
-        self.assertEqual(result, torch.eye(3, 3, device=device))
-        expected_aux = {'a': x.cos(), 'b': [x.tan()]}
-        self.assertEqual(aux, expected_aux)
+        # TODO: Remove when https://github.com/pytorch/functorch/issues/392
+        # is fixed
+        if jacapi == jacrev:
+            result, aux = jacapi(f, has_aux=True)(x)
+            self.assertEqual(result, torch.eye(3, 3, device=device))
+            expected_aux = {'a': x.cos(), 'b': [x.tan()]}
+            self.assertEqual(aux, expected_aux)
+        else:
+            result, aux = jacapi(f)(x)
+            self.assertEqual(result, torch.eye(3, 3, device=device))
+            expected_aux = {'a': -torch.diag(x.sin()), 'b': [torch.diag(1.0 + x.tan() ** 2)]}
+            self.assertEqual(aux, expected_aux)
 
-    @FIXME_jacrev_only
+    @jacrev_and_jacfwd
+    def test_outputs_can_any_pytree(self, device, jacapi):
+        x = torch.randn(2, 3, device=device)
+
+        for output in [None, ()]:
+            with self.assertRaisesRegex(RuntimeError, r"Expected f to be a function that has non-empty output"):
+                jacapi(lambda _: output)(x)
+
+        for output in [1, True, 12.2, "abc"]:
+            with self.assertRaisesRegex(RuntimeError, r"expected f\(\*primals\) to return only tensors"):
```

Comment:
Because it's a regex, we can actually check that it mentions some API, like (I don't remember if this is the syntax) "(vjp|jvp).*: Expected f to ..."